src/SMAIAXBackend.Application/DTOs/PolicyDto.cs

```diff
@@ -10,10 +12,15 @@ public class PolicyDto(
     LocationResolution locationResolution,
     decimal price)
 {
+    [Required]
     public Guid Id { get; set; } = id;
+    [Required]
```

Comment:
You can specify that `[Required]` does not allow empty strings with `[Required(AllowEmptyStrings=false)]`. It will then return a BadRequest by itself and you don't need to create a custom exception.